testsIT/src/test/scala/com/stratio/crossdata/driver/DriverIT.scala

```diff
@@ -29,155 +30,149 @@ import scala.reflect.io.File
-    val driver = Driver.getOrCreate()
-    val result = driver.addAppCommand(filePath, "com.stratio.addApp.AddAppTest.main", Some("testApp")).waitForResult()
-    driver.sql("EXECUTE testApp(rain,bow)").waitForResult()
-    driver.stop()
-    result.hasError should equal(false)
-
+    withDriverDo { driver =>
+      val result = driver.addAppCommand(filePath, "com.stratio.addApp.AddAppTest.main", Some("testApp")).waitForResult()
+      driver.sql("EXECUTE testApp(rain,bow)").waitForResult()
+      result.hasError should equal(false)
+    }
 
   }
 
   it should "be able to execute ADD APP Command of an existent file with options" in {
     assumeCrossdataUpAndRunning
 
     val filePath = getClass.getResource("/TestAddApp.jar").getPath
-    val driver = Driver.getOrCreate()
-    val addAppResult = driver.addAppCommand(filePath, "com.stratio.addApp.AddAppTest.main", Some("testApp")).waitForResult()
-    addAppResult.hasError should equal(false)
 
-    val executeResult = driver.sql("""EXECUTE testApp(rain,bow2) OPTIONS (executor.memory '20G')""").waitForResult()
+    withDriverDo { driver =>
+      val addAppResult = driver.addAppCommand(filePath, "com.stratio.addApp.AddAppTest.main", Some("testApp")).waitForResult()
+      addAppResult.hasError should equal(false)
+
+      val executeResult = driver.sql("""EXECUTE testApp(rain,bow2) OPTIONS (executor.memory '20G')""").waitForResult()
 
-    executeResult.hasError should equal(false)
-    executeResult.resultSet.length should equal(1)
-    executeResult.resultSet(0).get(0) should equal("Spark app launched")
+      executeResult.hasError should equal(false)
+      executeResult.resultSet.length should equal(1)
+      executeResult.resultSet(0).get(0) should equal("Spark app launched")
 
-    driver.stop()
+    }
   }
 
 
+  it should "allow running multiple drivers per JVM" in {
 
+    val driverTable = "drvtable"
+    val anotherDriverTable = "anotherTable"
 
+    withDriverDo { driver =>
+      withDriverDo { anotherDriver =>
+        driver shouldNot be theSameInstanceAs anotherDriver
+        driver.listTables().size shouldBe anotherDriver.listTables().size
 
+        driver.sql(s"CREATE TEMPORARY TABLE $driverTable USING org.apache.spark.sql.json OPTIONS (path '${Paths.get(getClass.getResource("/tabletest.json").toURI).toString}')").waitForResult()
+        driver.sql(s"SELECT * FROM $driverTable").waitForResult().resultSet should not be empty
+        anotherDriver.sql(s"SELECT * FROM $driverTable").waitForResult().hasError shouldBe true
 
+        anotherDriver.sql(s"CREATE TEMPORARY TABLE $anotherDriverTable USING org.apache.spark.sql.json OPTIONS (path '${Paths.get(getClass.getResource("/tabletest.json").toURI).toString}')").waitForResult()
+        anotherDriver.sql(s"SELECT * FROM $anotherDriverTable").waitForResult().resultSet should not be empty
+        driver.sql(s"SELECT * FROM $anotherDriverTable").waitForResult().hasError shouldBe true
+
+      }
+    }
+  }
 
```

Comment:
Maybe another test with 2 drivers to check if one driver close its session, the other driver still alive, and it can execute another query